### DISCUSSION TITLE
/articles/microsoft-365-apps-use-spfx-or-microsoft-teams/

### DISCUSSION THREAD

**Discussion by giscus** [Voitanos/voitanos.io-comments#234] [Comments]
Developers can use the SharePoint Framework or Microsoft Teams apps for a new Microsoft 365 application. Compare the pros and cons of in this article.
https://www.andrewconnell.com/articles/microsoft-365-apps-use-spfx-or-microsoft-teams/

**Comment by rvhelden** [comment]
Is the teams first model affected by this announcement? As it seems to use a provider hosted add-in / webpart.
https://www.voitanos.io/blog/sharepoint-add-in-model-retirement-other-services-unpacked/

**Reply by andrewconnell** [reply]
Nope... not a bit. It doesn't use the add-in infrastructure at all. 

The web app that's surfaced as a tab in Microsoft Teams lives in a cloud resource (let's say Azure Web App, but it can be web hosting option). When rendered in SharePoint, an `iframe` is used to load the same page, just like it's done in Microsoft Teams.

**Comment by anthonywhite** [comment]
Hi Andrew, great article. Just want to check the [Teams App in SharePoint](https://www.voitanos.io/blog/microsoft-365-apps-use-spfx-or-microsoft-teams/#deploy-microsoft-teams-app-to-sharepoint-online) scenario - can the Teams apps be added to either Tenant or Site Collection App Catalogs? And is the tenant-scope deploy option supported?

**Reply by andrewconnell** [reply]
It should work in either scenario (tenant-scoped or site collection-scoped). Unlike the SPFx => Teams option, when you use the Teams App => SPO, you have to manually upload the Teams App Package to the app catalog. 

Because the app catalog infrastructure under the covers is essentially the same, it shouldn't matter.

**Comment by amxMTerbeck** [comment]
Hello Andrew, 
I wanted to try the Teams => SPO Online Solution. With the Teams Toolkit I created a Tab-Projekt and added configurableTabs in the manifest.json.

Afterwards I uploaded the package to the SharePoint App Catalog and it correctly asked if I wanted to add the app to all sites. When I now want to add the WebPart to a page or create an app page it correctly shows the solution inside the webpart selection, but after selection nothing happens the page keeps empty and event in the dev tools there is nothing hinting to an error.

Do I miss something? I already googled the behaviour but couldn't find any solution:
https://github.com/MicrosoftDocs/msteams-docs/issues/7579 
https://github.com/OfficeDev/microsoft-teams-library-js/issues/1851


    "configurableTabs": [
        {
            "configurationUrl": "https://tabdf4192.azurewebsites.net/tab",
            "canUpdateConfiguration": true,
            "scopes": [
                "team"
            ],
            "supportedSharePointHosts": [
                "sharePointFullPage",
                "sharePointWebPart"
            ]
        }
    ],

**Reply by andrewconnell** [reply]
Looking at those issues, they seem to indicate the error not with Teams, but rather on the SPO side.

1. Have you opened a bug in the [SharePoint/sp-dev-docs](https://github.com/SharePoint/sp-dev-docs) repo? I don't see anything in the repo... maybe I'm missing it?
1. When you say:

    > ... the page keeps empty and event in the dev tools there is nothing hinting to an error.

    Are you saying there's nothing in the browser's dev tools showing an error? Nothing in the **Console** or **Network** tabs? Do you see a request in the **Network** tab to the URL of your tab?

If you haven't opened a bug in the [SharePoint/sp-dev-docs](https://github.com/SharePoint/sp-dev-docs) repo's issue list, I'd do that first. When you do, please drop a link in the comments here so others can track it. I'll also make sure the SharePoint engineering team is aware of it.

**Reply by amxMTerbeck** [reply]
Thank you, I will try it in the sp docs :)
